libvast/CMakeLists.txt

```diff
@@ -311,24 +311,33 @@ if (VAST_ENABLE_RELOCATABLE_INSTALLATIONS
           DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 endif ()
 
-# -- source files --------------------------------------------------------------
+# -- libvast -------------------------------------------------------------------
 
 file(GLOB_RECURSE libvast_sources CONFIGURE_DEPENDS
      "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp"
      "${CMAKE_CURRENT_SOURCE_DIR}/vast/*.hpp")
 
+list(APPEND libvast_sources
+  "${CMAKE_CURRENT_BINARY_DIR}/vast/config.hpp"
+  "${CMAKE_CURRENT_BINARY_DIR}/src/config.cpp")
+
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/vast/config.hpp.in"
                "${CMAKE_CURRENT_BINARY_DIR}/vast/config.hpp")
-configure_file("${CMAKE_CURRENT_SOURCE_DIR}/src/config.cpp.in"
-               "${CMAKE_CURRENT_BINARY_DIR}/src/config.cpp")
-
-list(APPEND libvast_sources "${CMAKE_CURRENT_BINARY_DIR}/vast/config.hpp"
-     "${CMAKE_CURRENT_BINARY_DIR}/src/config.cpp")
-
-# -- libvast -------------------------------------------------------------------
 
 add_library(libvast ${libvast_sources})
 add_library(vast::libvast ALIAS libvast)
+
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/update-config.cmake" "\
+  get_filename_component(CMAKE_MODULE_PATH
+    \"${CMAKE_CURRENT_SOURCE_DIR}/../cmake\" ABSOLUTE)
+  include(VASTGitVersion)
+  configure_file(\"${CMAKE_CURRENT_SOURCE_DIR}/src/config.cpp.in\"
+                 \"${CMAKE_CURRENT_BINARY_DIR}/src/config.cpp\")")
+
+add_custom_target(libvast_update_config
+  BYPRODUCTS "${CMAKE_CURRENT_BINARY_DIR}/src/config.cpp"
+  COMMAND ${CMAKE_COMMAND} -P "${CMAKE_CURRENT_BINARY_DIR}/update-config.cmake")
```

Comment:
This new indirection has to be documented in a comment that explicitly states which problem it solves and why this method is appropriate.

Reply:
Fair point, a command is definitely needed here.

Reply:
Done